pkg/csilvm/arbiter.go

```diff
@@ -0,0 +1,422 @@
+package csilvm
+
+import (
+	"context"
+	"encoding/json"
+	"sort"
+	"sync"
+
+	"github.com/container-storage-interface/spec/lib/go/csi/v0"
+	"github.com/mesosphere/csilvm/pkg/singleflight"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+)
+
+var _ = csi.IdentityServer(&identityArbiter{}) // sanity check
+
+type RPCCounters struct {
+	IssuedCall func() int64 // callback useful for testing
+	SharedCall func() int64 // callback useful for testing
+}
+
+type GenericArbiter struct {
+	RPCCounters
+	singleflight.Group
```

Comment:
We might need to share `singleflight.Group` across arbiter instances if, for example, all ops for the same `volume_id` should not overlap.